File: Codes/stack_and_queue/Stack.java
public class Stack {

    int size = -1;
    private final int DEFAULT_SIZE = 10;
    private int[] arr = new int[DEFAULT_SIZE];
    private int currentSize = DEFAULT_SIZE;

    public void push(int num) {
        if (currentSize == (size + 1)) {
            currentSize = currentSize * 2;
            int[] temp = new int[currentSize];
            for (int i = 0; i <= size; i++) {
                temp[i] = arr[i];
            }
            arr = temp;
        }
        arr[++size] = num;
    }

    public void print() {
        for (int i = 0; i <= size; i++) {
            System.out.print(arr[i] + "\t");
        }
    }
}

File: Codes/stack_and_queue/Stack.kt
class Stack {
    private var DEFAULT_SIZE = 1
    var stackArr = Array(DEFAULT_SIZE){0}
    var stackSize = 0
    fun push(num: Int) {
        if ((DEFAULT_SIZE - 1) < stackSize) increaseStackSize()
        this.stackArr[stackSize] = num
        this.stackSize++
    }

    fun increaseStackSize() {
        this.DEFAULT_SIZE = DEFAULT_SIZE * 2
        var newArr = Array(DEFAULT_SIZE){0}
        for (i in 0 until stackSize) {
            newArr[i] = stackArr[i]
        }
        this.stackArr = newArr
    }

    fun print() {
        for (i in 0 until stackSize) {
            println(stackArr[i])
        }
    }
}

fun main() {
    val stack = Stack()
    stack.push(1)
    stack.push(2)
    stack.push(3)
    stack.push(4)
    stack.push(5)
    stack.push(6)

    stack.print()
}
